plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLQueryCompletionAnalyzer.java

```diff
@@ -214,12 +215,20 @@ private DBPImage prepareProposalImage(@NotNull SQLQueryCompletionItem item) {
             case RESERVED -> UIIcon.SQL_TEXT;
             case SUBQUERY_ALIAS -> DBIcon.TREE_TABLE_ALIAS;
             case DERIVED_COLUMN_NAME -> DBIcon.TREE_FOREIGN_KEY_COLUMN;
-            case NEW_TABLE_NAME -> DBIcon.TREE_TABLE;
-            case USED_TABLE_NAME -> UIIcon.EDIT_TABLE;
+            case NEW_TABLE_NAME -> this.prepareTableProposalIcon(item.getObject(), false);
+            case USED_TABLE_NAME -> this.prepareTableProposalIcon(item.getObject(), true);
             case TABLE_COLUMN_NAME -> DBIcon.TREE_COLUMN;
             case JOIN_CONDITION -> DBIcon.TREE_CONSTRAINT;
             default -> throw new IllegalStateException("Unexpected completion item kind " + item.getKind());
         };
         return image;
     }
+
+    private DBPImage prepareTableProposalIcon(DBPObject object, boolean used) {
+        DBPImage icon = object == null ? DBIcon.TREE_TABLE : DBValueFormatting.getObjectImage(object);
+        if (used) {
+            icon = new DBIconComposite(icon, false, null, DBIcon.OVER_SUCCESS, null, null);
```

Comment:
Why OVER icon for tables?

Reply:
We decided to remove this feature and discuss first